src/tools/rebuild_mondb.cc

```diff
@@ -277,6 +277,17 @@ int update_osdmap(ObjectStore& fs, OSDSuperblock& sb, MonitorDBStore& ms)
           // inc.decode() verifies `inc_crc`, so it's been taken care of.
         }
       }
+    }();
```

Comment:
@tchaikov This commit changed this line from `}` to `}();`, which is now causing ceph master build to fail in the OBS with:

```
[ 6997s] I: Program returns random data in a function
[ 6997s] E: ceph no-return-in-nonvoid-function /home/abuild/rpmbuild/BUILD/ceph-15.0.0-3913-gb9c7443/src/tools/rebuild_mondb.cc:280
```

(The OBS does some static code analysis on all builds...)

Why is the `()` part needed here? And can you do something to appease the OBS static code analysis?

Reply:
Just to clarify my previous comment:

I assume the `()` was added to turn the `{ . . . }` block into an unnamed function, so it can be called and the return value assigned to the variable `add_inc_result`.

Perhaps the static analyzer thinks there is a possibility that the function might return an undefined pointer value, potentially creating an attack vector?

Reply:
@smithfarm the static analyzer is correct. we need https://github.com/ceph/ceph/pull/29704 also, see https://tracker.ceph.com/issues/41177#note-5